### ISSUE TITLE
False positive `no-name-in-module` when importing from ``from ccxt.base.errors`` even when using the ``ignored-modules`` option

### ISSUE THREAD

**Issue by eabrouwer3** [PyCQA/pylint#8148]
### Bug description

Simply importing exceptions from the [`ccxt`](https://github.com/ccxt/ccxt) library is giving this error. Here's an example of how we import them:

```python
from ccxt.base.errors import (
    AuthenticationError,
    ExchangeError,
    ExchangeNotAvailable,
    NetworkError,
    RateLimitExceeded,
    RequestTimeout,
)
```

Pycharm can find the exception classes just fine. I know they exist. It could have something to do with how the library is using `__all__`, but I don't know too much about how that works to draw that conclusion.

Also, note that we're using version 1.95.1 of `ccxt`. We use it in some critical paths, so we can't update it to the latest version quite yet.

The configuration written below is what I've tried, but it seems based on googling that that doesn't stop all errors from being ignored regarding those modules. So I'm still getting the issue.

### Configuration

```ini
# List of module names for which member attributes should not be checked
# (useful for modules/projects where namespaces are manipulated during runtime
# and thus existing member attributes cannot be deduced by static analysis). It
# supports qualified module names, as well as Unix pattern matching.
ignored-modules=ccxt,ccxt.base,ccxt.base.errors
```

### Command used

```shell
pylint test_ccxt_base_errors.py
```

### Pylint output

```shell
************* Module test_ccxt_base_errors
test_ccxt_base_errors.py:1:0: E0611: No name 'errors' in module 'list' (no-name-in-module)
```

### Expected behavior

No error to be reported

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.9.16 (main, Dec  7 2022, 10:16:11)
[Clang 14.0.0 (clang-1400.0.29.202)]
```

### OS / Environment

Intel based 2019 Mac Book Pro. Mac OS 13.1 (Ventura). Fish shell.

### Additional dependencies

ccxt==1.95.1

**Comment by Pierre-Sassoulas** [comment]
Could you upgrade to at least 2.15.10 (better yet 2.16.0b1) and confirm the issue still exists, please ?

**Comment by eabrouwer3** [comment]
Tried with

```
pylint 2.15.10
astroid 2.13.4
Python 3.9.16 (main, Dec  7 2022, 10:16:11)
[Clang 14.0.0 (clang-1400.0.29.202)]
```
and also with pylint 2.16.0b1 and I still get the same issue.

**Comment by Pierre-Sassoulas** [comment]
Thank you ! I can reproduce, and ``ignored-modules`` does work with a simpler example like ``random.foo``

**Comment by clavedeluna** [comment]
@Pierre-Sassoulas is the fix here:
1. figure out why the ccxt library causes a `no-name-in-module` msg
2. figure out why using `ignored-modules` is still raising `no-name-in-module`
?

**Comment by Pierre-Sassoulas** [comment]
Yes, I think 2/ is the one to prioritize as it's going to be useful for everyone and not just ccxt users. But if we manage find the root cause of 1/ it's going to be generic too.

**Comment by clavedeluna** [comment]
I think there's a bug in variables.py:_check_module_attrs, I don't think `module = next(module.getattr(name)[0].infer())` should be reassigning to module. Will continue to investigate.

**Comment by clavedeluna** [comment]
There is a non-ccxt root cause. This issue can be reproduced with the following dir structure:

```
pkg_mod_imports/__init__.py
pkg_mod_imports/base/__init__.py
pkg_mod_imports/base/errors.py
```
pkg_mod_imports/__init__.py should have :
```
base = [
    'Exchange',
    'Precise',
    'exchanges',
    'decimal_to_precision',
]
```

and  pkg_mod_imports/base/errors.py

```
class SomeError(Exception):
    pass
```

in a test.py module add
```
from pkg_mod_imports.base.errors import SomeError
```
And then running `pylint test.py` the result is
```
test.py:1:0: E0611: No name 'errors' in module 'list' (no-name-in-module)
```

It's coming from the fact that `errors` is both a list inside the init file and the name of a module. variable.py does `module = next(module.getattr(name)[0].infer())` . `getattr` fetches the `errors` list, not the module!
